samplers/aws/xray/remote_sampler.go

```diff
@@ -0,0 +1,219 @@
+	}
+}
+
+func (rs *remoteSampler) startRulePoller(ctx context.Context) {
+	go func() {
+		// Period = 300s, Jitter = 5s
+		t := newTicker(rs.samplingRulesPollingInterval, 5*time.Second)
+
+		// Periodic manifest refresh
+		for {
+			if err := rs.refreshManifest(ctx); err != nil {
+				globalLogger.Error(err, "Error occurred while refreshing sampling rules")
+			} else {
+				globalLogger.Info("Successfully fetched sampling rules")
+			}
+			select {
+			case _, more := <-t.C():
+				if !more {
+					return
+				}
+				continue
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+}
+
+func (rs *remoteSampler) refreshManifest(ctx context.Context) (err error) {
+	// Compute 'now' before calling GetSamplingRules to avoid marking manifest as
+	// fresher than it actually is.
+	now := rs.clock.now().Unix()
+
+	// Get sampling rules from proxy.
+	rules, err := rs.xrayClient.getSamplingRules(ctx)
+	if err != nil {
+		return
+	}
+
+	failed := false
+
+	// temporary manifest declaration.
+	tempManifest := &manifest{
+		rules: []*rule{},
+		index: map[string]*rule{},
+		clock: &defaultClock{},
+	}
+
+	for _, records := range rules.SamplingRuleRecords {
+		if records.SamplingRule.RuleName == nil {
+			globalLogger.V(1).Info("Sampling rule without rule name is not supported")
+			failed = true
+			continue
+		}
+
+		// Only sampling rule with version 1 is valid
+		if records.SamplingRule.Version == nil {
+			globalLogger.V(1).Info("Sampling rule without version number is not supported", "RuleName", *records.SamplingRule.RuleName)
+			failed = true
+			continue
+		}
+
+		if *records.SamplingRule.Version != int64(1) {
+			globalLogger.V(1).Info("Sampling rule without version 1 is not supported", "RuleName", *records.SamplingRule.RuleName)
+			failed = true
+			continue
+		}
+
+		if len(records.SamplingRule.Attributes) != 0 {
+			globalLogger.V(1).Info("Sampling rule with non nil Attributes is not applicable", "RuleName", *records.SamplingRule.RuleName)
+			continue
+		}
+
+		if records.SamplingRule.ResourceARN == nil {
+			globalLogger.V(1).Info("Sampling rule without ResourceARN is not applicable", "RuleName", *records.SamplingRule.RuleName)
+			continue
+		}
+
+		if *records.SamplingRule.ResourceARN != "*" {
+			globalLogger.V(1).Info("Sampling rule with ResourceARN not equal to * is not applicable", "RuleName", *records.SamplingRule.RuleName)
+			continue
+		}
+
+		// create rule and store it in temporary manifest to avoid locking issues.
+		createErr := tempManifest.createRule(records.SamplingRule)
+		if createErr != nil {
+			failed = true
+			globalLogger.Error(createErr, "Error occurred creating/updating rule")
+		}
+	}
+
+	// Set err if updates failed
+	if failed {
+		err = errors.New("error occurred creating/updating rules")
+	}
+
+	// Re-sort to fix matching priorities.
+	tempManifest.sort()
+	// Update refreshedAt timestamp
+	tempManifest.refreshedAt = now
+
+	// assign temp manifest to original copy/one sync refresh.
+	rs.mu.Lock()
```

Comment:
This might be for another PR, but wondering if volatile write / read is an appropriate use for `atomic` in Go too, but not sure